.github/workflows/build-test-publish-on-push-cached.yaml

```diff
@@ -43,7 +43,7 @@ jobs:
       - uses: pnpm/action-setup@v3
         with:
           version: 8
-      - run: pnpm add -g pnpm
+      - run: pnpm add -g pnpm@8.15.7
```

Comment:
do we need to stick to a specific pnpm version here?

Reply:
It's the last version of pnpm that supports npm16, but no longer needed due to #207